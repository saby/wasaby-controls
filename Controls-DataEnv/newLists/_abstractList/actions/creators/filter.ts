/**
 * @kaizen_zone 039c82f1-a0a3-4548-82d6-c9e1dbaf5de0
 */
import type { TFilter } from 'Controls-DataEnv/interface';
import type { filter } from '../types';

/**
 * Конструктор действия для установки нового фильтра.
 * @function
 * @param {TFilter} filter Фильтр
 * @return filter.TSetFilterAction
 */
export const setFilter = (filter: TFilter): filter.TSetFilterAction => ({
    type: 'setFilter',
    payload: {
        filter,
    },
});

/**
 * Конструктор действия для открытия окон фильтров.
 * @function
 * @return filter.TOpenFilterDetailPanelAction
 */
export const openFilterDetailPanel = (): filter.TOpenFilterDetailPanelAction => ({
    type: 'openFilterDetailPanel',
    payload: {},
});

/**
 * Конструктор действия для закрытия окон фильтров.
 * @function
 * @return filter.TCloseFilterDetailPanelAction
 */
export const closeFilterDetailPanel = (): filter.TCloseFilterDetailPanelAction => ({
    type: 'closeFilterDetailPanel',
    payload: {},
});
