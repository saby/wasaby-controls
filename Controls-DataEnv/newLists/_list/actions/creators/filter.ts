/**
 * @kaizen_zone 039c82f1-a0a3-4548-82d6-c9e1dbaf5de0
 */
import type { IListState } from '../../interface/IListState';
import type { filter } from '../types';

/**
 * Конструктор действия, для установки структуры фильтров.
 */
export const setFilterDescription = ({
    filterDescription,
    countFilterValue,
    countFilterLinkedNames,
    countFilterValueConverter,
    countFilterUserPeriods,
    countFilterPeriodType,
}: Pick<
    filter.IFilterState,
    | 'filterDescription'
    | 'countFilterValue'
    | 'countFilterLinkedNames'
    | 'countFilterValueConverter'
    | 'countFilterUserPeriods'
    | 'countFilterPeriodType'
>): filter.TSetFilterDescriptionAction => ({
    type: 'setFilterDescription',
    payload: {
        filterDescription,
        countFilterValue,
        countFilterLinkedNames,
        countFilterValueConverter,
        countFilterUserPeriods,
        countFilterPeriodType,
    },
});

/**
 * Конструктор действия, для обновления фильтра.
 */
export const updateFilter = (
    prevState: IListState,
    {
        filter,
        countFilterValue,
        filterDescription,
        countFilterValueConverter,
        countFilterLinkedNames,
    }: filter.IFilterState
): filter.TUpdateFilterAction => ({
    type: 'updateFilter',
    payload: {
        prevState,
        filter,
        countFilterValue,
        filterDescription,
        countFilterValueConverter,
        countFilterLinkedNames,
    },
});
