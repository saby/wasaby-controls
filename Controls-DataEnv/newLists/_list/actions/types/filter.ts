/**
 * @kaizen_zone 039c82f1-a0a3-4548-82d6-c9e1dbaf5de0
 */
import type { TAbstractAction } from 'Controls-DataEnv/dispatcher';
import type { TFilter } from 'Controls-DataEnv/interface';
import type { IListState } from '../../interface/IListState';
import type { IFilterItem, IUserPeriod } from 'Controls/filter';
import type { QueryWhereExpression } from 'Types/source';
import type { TAbstractListActions } from 'Controls-DataEnv/abstractList';

// FIXME: Убрать этот тип. Минимум - из export, максимум - вообще.
/**
 * IFilterState
 */
export interface IFilterState {
    filter?: TFilter;
    filterDescription?: IFilterItem[];
    countFilterValue?: string | Date[];
    countFilterLinkedNames?: string[];
    countFilterValueConverter?: (
        value: string | Date | Date[],
        filterItem: IFilterItem,
        filterDescription: IFilterItem[]
    ) => QueryWhereExpression<unknown>;
    countFilterUserPeriods?: IUserPeriod[];
    countFilterPeriodType?: string;
}

//# region Экспорты для публичных типов.
/**
 * Тип действия, для установки нового фильтра.
 */
export type TSetFilterAction = TAbstractListActions.filter.TSetFilterAction;
/**
 * Тип действия для открытия окон фильтров.
 */
export type TOpenFilterDetailPanelAction = TAbstractListActions.filter.TOpenFilterDetailPanelAction;
/**
 * Тип действия для закрытия окон фильтров.
 */
export type TCloseFilterDetailPanelAction =
    TAbstractListActions.filter.TCloseFilterDetailPanelAction;
//# endregion Экспорты для публичных типов.

/**
 * Тип действия, для установки структуры фильтров.
 */
export type TSetFilterDescriptionAction = TAbstractAction<
    'setFilterDescription',
    Pick<
        IFilterState,
        | 'filterDescription'
        | 'countFilterValue'
        | 'countFilterLinkedNames'
        | 'countFilterValueConverter'
        | 'countFilterUserPeriods'
        | 'countFilterPeriodType'
    >
>;

/**
 * Тип действия, для обновления фильтра.
 */
export type TUpdateFilterAction = TAbstractAction<
    'updateFilter',
    {
        prevState: IListState;
    } & IFilterState
>;

/**
 * Тип действий функционала "Фильтрация", доступные в WEB списке.
 * @see https://online.sbis.ru/area/849d2ba6-201e-467e-ae1a-d32fca6084bd Зона Kaizen
 */
export type TAnyFilterAction =
    | TAbstractListActions.filter.TAnyFilterAction
    | TSetFilterDescriptionAction
    | TUpdateFilterAction;
