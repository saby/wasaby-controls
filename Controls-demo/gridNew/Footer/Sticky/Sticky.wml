<div class="controlsDemo__wrapper controlDemo__grid-header-sticky">
    <Controls.scroll:Container attr:class="controlsDemo__height400">
        <Controls.grid:View
                keyProperty="key"
                source="{{_viewSource}}"
                stickyFooter="{{true}}"
                columns="{{_columns}}">
            <ws:footerTemplate>
                <ws:partial template="Controls/grid:FooterTemplate">
                    <div>Контент подвала таблицы</div>
                </ws:partial>
            </ws:footerTemplate>
        </Controls.grid:View>
    </Controls.scroll:Container>
</div>
