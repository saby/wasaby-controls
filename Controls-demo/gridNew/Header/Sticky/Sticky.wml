<div class="controlsDemo__wrapper controlDemo__grid-header-sticky">
    <Controls.scroll:Container attr:class="controlsDemo__height400 controlsDemo__width800px">
        <Controls.grid:View
                backgroundStyle="default"
                storeId="HeaderSticky"
                header="{{_header}}"
                columns="{{_columns}}"/>
    </Controls.scroll:Container>
</div>
