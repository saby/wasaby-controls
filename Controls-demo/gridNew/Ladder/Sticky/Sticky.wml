<div class="controlsDemo__wrapper">
    <Controls.scroll:Container attr:class="controlsDemo__height400" shadowMode="js">
        <Controls.listDeprecate:Mover name="listMover"/>
        <Controls.grid:View
				name="grid"
                keyProperty="key"
                source="{{_viewSource}}"
                columns="{{_columns}}"
                itemsDragNDrop="{{true}}"
                on:customdragStart="_dragStart()"
                on:customdragEnd="_dragEnd()"
                multiSelectVisibility="visible"
                bind:selectedKeys="_selectedKeys"
                header="{{_header}}"
                resultsPosition="top"
                ladderProperties="{{_ladderProperties}}">
			<ws:footerTemplate>
				<ws:partial template="Controls/grid:FooterTemplate">
					<Controls.list:AddButton caption="Add record" on:click="_beginAdd()" data-qa="controlsDemo-GridLadder__addButton"/>
				</ws:partial>
			</ws:footerTemplate>
		</Controls.grid:View>
    </Controls.scroll:Container>
</div>
