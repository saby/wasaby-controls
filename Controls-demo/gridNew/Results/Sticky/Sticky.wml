<div class="controlsDemo__wrapper controlsDemo_fixedWidth800">
    <Controls.scroll:Container attr:class="controlsDemo__height400">
        <Controls.grid:View
                backgroundStyle="default"
                storeId="ResultsSticky"
                header="{{_header}}"
                resultsPosition="top"
                columns="{{_columns}}"/>
    </Controls.scroll:Container>
</div>
