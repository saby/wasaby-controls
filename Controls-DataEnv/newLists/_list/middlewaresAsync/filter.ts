/**
 * @kaizen_zone 039c82f1-a0a3-4548-82d6-c9e1dbaf5de0
 */
import { asyncMiddlewareFactory } from 'Controls-DataEnv/dispatcher';
import { TListMiddleware } from '../types/TListMiddleware';

export const filterMiddleware: TListMiddleware = asyncMiddlewareFactory(
    'Controls/listWebReducers:filter',
    'filter',
    ['setFilterDescription', 'setFilter', 'updateFilter']
);
