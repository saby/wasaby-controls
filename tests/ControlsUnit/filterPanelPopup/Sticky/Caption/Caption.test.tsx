/**
 * @jest-environment jsdom
 */
import { unmountComponentAtNode } from 'react-dom';
import { Sticky } from 'Controls/filterPanelPopup';
import { TFilterItemViewMode } from 'Controls/filter';
import { render } from '@testing-library/react';
import { WasabyEvents } from 'UICore/Events';
import 'ControlsUnit/filterPanelPopup/Sticky/EmptyEditorForTests';

function getPanel(container: HTMLElement): HTMLElement {
    return container.querySelector('.controls-FilterViewPanel');
}

describe('Controls/filterPanelPopup:Sticky caption tests', () => {
    let container = null;

    beforeEach(() => {
        container = document.createElement('div');
        WasabyEvents.initInstance(container);
        document.body.appendChild(container);
    });
    afterEach(() => {
        unmountComponentAtNode(container);
        WasabyEvents.destroyInstance(container);
        container.remove();
        container = null;
    });

    describe('Тест меток редакторов', () => {
        it('Метки в блоке "Отбираются", редактор с меткой и редактор без метки', () => {
            const filterDescription = [
                {
                    name: 'withoutCaption',
                    viewMode: 'basic' as TFilterItemViewMode,
                    value: [null],
                    resetValue: [null],
                    editorTemplateName:
                        'ControlsUnit/filterPanelPopup/Sticky/EmptyEditorForTests',
                },
                {
                    name: 'withCaption',
                    caption: 'Метка',
                    viewMode: 'basic' as TFilterItemViewMode,
                    value: [null],
                    resetValue: [null],
                    editorTemplateName:
                        'ControlsUnit/filterPanelPopup/Sticky/EmptyEditorForTests',
                },
            ];

            render(<Sticky items={filterDescription} />, { container });
            expect(getPanel(container)).toMatchSnapshot();
        });

        it('Метка в блоке "Еще можно отобрать"', () => {
            const filterDescription = [
                {
                    name: 'withCaption',
                    extendedCaption: 'Метка',
                    viewMode: 'extended' as TFilterItemViewMode,
                    value: [null],
                    resetValue: [null],
                    editorTemplateName:
                        'ControlsUnit/filterPanelPopup/Sticky/EmptyEditorForTests',
                },
            ];

            render(<Sticky items={filterDescription} />, { container });
            expect(getPanel(container)).toMatchSnapshot();
        });

        describe('Разный текст метки при переходе из блока "Еще можно отобрать" в блок "Отбираются"', () => {
            it('Редактор в блоке "Отбираются"', () => {
                const filterDescription = [
                    {
                        name: 'withCaption',
                        caption: 'Метка для блока "Отбираются"',
                        extendedCaption: 'Метка для блока "Ещё можно отобрать"',
                        viewMode: 'basic' as TFilterItemViewMode,
                        value: [null],
                        resetValue: ['test'],
                        editorTemplateName:
                            'ControlsUnit/filterPanelPopup/Sticky/EmptyEditorForTests',
                    },
                ];

                render(<Sticky items={filterDescription} />, { container });
                expect(getPanel(container)).toMatchSnapshot();
            });
            it('Редактор в блоке "Ещё можно отобрать"', () => {
                const filterDescription = [
                    {
                        name: 'withCaption',
                        caption: 'Метка для блока "Отбираются"',
                        extendedCaption: 'Метка для блока "Ещё можно отобрать"',
                        viewMode: 'extended' as TFilterItemViewMode,
                        value: [null],
                        resetValue: [null],
                        editorTemplateName:
                            'ControlsUnit/filterPanelPopup/Sticky/EmptyEditorForTests',
                    },
                ];

                render(<Sticky items={filterDescription} />, { container });
                expect(getPanel(container)).toMatchSnapshot();
            });
        });
    });
});
