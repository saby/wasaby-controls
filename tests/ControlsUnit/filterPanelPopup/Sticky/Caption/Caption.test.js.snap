// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls/filterPanelPopup:Sticky caption tests Тест меток редакторов Метка в блоке "Еще можно отобрать" 1`] = `
<div
  class="controls_filterPanel_theme-default  controls-FilterViewPanel-contrast   controls_filter_theme-default controls-FilterViewPanel controls-Scroll-containerBase_userContent"
  data-qa="controls-FilterPanelPopup__sticky_filterPanel"
  tabindex="0"
>
  <div
    class=""
    tabindex="0"
  >
    <div
      class="controls-FilterViewPanel__editors-container controls-padding_bottom-l controls-FilterViewPanel__horizontal-padding-popup controls-FilterViewPanel__vertical-padding-popup controls-FilterViewPanel__editors-container-popup     controls-FilterPanel__extendedItems-padding-top"
      data-qa="controls-FilterViewPanel__additional-editors"
      tabindex="0"
    >
      <div
        class="controls-FilterViewPanel__additional-editors controls-FilterViewPanel__additional-editors-popup controls-FilterViewPanel__additional-editors-in-column"
        tabindex="0"
      >
        <div
          class="controls-FilterViewPanel__additional_left-column controls-FilterViewPanel__additional_left-column-maxWidth"
          data-qa="controls-FilterViewPanel__additional_left-column"
        >
          <div
            class="controls-FilterViewPanel__additional-editor controls-FilterViewPanel__additional-editor_default controls-fontsize-m controls-text-filterExtendedItem controls-FilterViewPanel__additional-editor-popup_last controls-FilterViewPanel__additional-editor-popup-in-column"
            data-name="FilterViewPanel__additional-editor_withCaption"
            data-qa="FilterViewPanel__additional-editor"
            title="Метка"
          >
            Метка
          </div>
        </div>
        <div
          class="controls-FilterViewPanel__additional_right-column"
          data-qa="controls-FilterViewPanel__additional_right-column"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Controls/filterPanelPopup:Sticky caption tests Тест меток редакторов Метки в блоке "Отбираются", редактор с меткой и редактор без метки 1`] = `
<div
  class="controls_filterPanel_theme-default  controls-FilterViewPanel-contrast   controls_filter_theme-default controls-FilterViewPanel controls-Scroll-containerBase_userContent"
  data-qa="controls-FilterPanelPopup__sticky_filterPanel"
  tabindex="0"
>
  <div
    class="controls-FilterViewPanel_viewMode-popup-vertical  controls-FilterViewPanel-contrast"
    data-qa="controls-FilterViewPanel__basic-editors"
    tabindex="0"
  >
    <div
      class="controls-PropertyGrid controls_propertyGrid_theme-default controls_list_theme-default controls_toggle_theme-default controls-PropertyGrid__itemsContainer_paddingTop-null controls-PropertyGrid__itemsContainer_paddingBottom-null controls-PropertyGrid__itemsContainer_paddingLeft-filterPanelPopup controls-PropertyGrid__itemsContainer_paddingRight-l controls-FilterViewPanel__propertyGrid"
      data-qa="FilterViewPanel__propertyGrid"
      tabindex="0"
    >
      <div
        attr-data-qa="controls-FilterViewPanel__item_withoutCaption"
        class="controls-PropertyGrid__editor    controls-PropertyGrid__editor_spacingTop_default
                       controls-PropertyGrid__editor_spacingBottom_null controls-PropertyGrid__editor_spacingRight_default controls-PropertyGrid__editor_spacingLeft_default controls-FilterViewPanel__propertyGrid_itemTemplate
                controls-FilterViewPanel__propertyGrid_itemTemplate_viewMode-popup-vertical"
        data-qa="controls-PropertyGrid__editor"
        item-key="property-grid-item-withoutCaption"
        tabindex="1"
      >
        <div
          class="controls-PropertyGrid__editor-withoutCaption controls-FilterViewPanel__baseEditor-container controls-FilterViewPanel__baseEditor-container_default"
          data-qa="controls-PropertyGrid__editor_withoutCaption"
        >
          <span />
        </div>
      </div>
      <div
        class="controls-FilterViewPanel__group-wrapper controls-FilterViewPanel__group-wrapper_style-master controls-FilterViewPanel__group-wrapper-margin-top"
        data-qa="FilterViewPanel__group-wrapper"
        item-key="group-withCaption"
        tabindex="2"
      >
        <div
          class="controls-FilterViewPanel__separator-content_container controls-FilterViewPanel__separator-content_container_popup ws-flexbox controls-FilterViewPanel__group controls-FilterViewPanel__group_height_viewMode-popup"
          data-qa="FilterViewPanel__group"
          tabindex="0"
        >
          <div
            class="controls-FilterViewPanel__group-content controls-FilterViewPanel__group-content_popup  controls-FilterViewPanel__group-content-align-left"
            tabindex="0"
            title="Метка"
          >
            Метка
          </div>
        </div>
      </div>
      <div
        attr-data-qa="controls-FilterViewPanel__item_withCaption"
        class="controls-PropertyGrid__editor    controls-PropertyGrid__editor_spacingTop_default
                       controls-PropertyGrid__editor_spacingBottom_null controls-PropertyGrid__editor_spacingRight_default controls-PropertyGrid__editor_spacingLeft_default controls-FilterViewPanel__propertyGrid_itemTemplate
                controls-FilterViewPanel__propertyGrid_itemTemplate_viewMode-popup-vertical_last-without-extended-items controls-FilterViewPanel__propertyGrid_itemTemplate_viewMode-popup_with-editorCaption"
        data-qa="controls-PropertyGrid__editor"
        item-key="property-grid-item-withCaption"
        tabindex="3"
      >
        <div
          class="controls-PropertyGrid__editor-withoutCaption controls-FilterViewPanel__baseEditor-container controls-FilterViewPanel__baseEditor-container_default"
          data-qa="controls-PropertyGrid__editor_withCaption"
        >
          <span />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Controls/filterPanelPopup:Sticky caption tests Тест меток редакторов Разный текст метки при переходе из блока "Еще можно отобрать" в блок "Отбираются" Редактор в блоке "Ещё можно отобрать" 1`] = `
<div
  class="controls_filterPanel_theme-default  controls-FilterViewPanel-contrast   controls_filter_theme-default controls-FilterViewPanel controls-Scroll-containerBase_userContent"
  data-qa="controls-FilterPanelPopup__sticky_filterPanel"
  tabindex="0"
>
  <div
    class=""
    tabindex="0"
  >
    <div
      class="controls-FilterViewPanel__editors-container controls-padding_bottom-l controls-FilterViewPanel__horizontal-padding-popup controls-FilterViewPanel__vertical-padding-popup controls-FilterViewPanel__editors-container-popup     controls-FilterPanel__extendedItems-padding-top"
      data-qa="controls-FilterViewPanel__additional-editors"
      tabindex="0"
    >
      <div
        class="controls-FilterViewPanel__additional-editors controls-FilterViewPanel__additional-editors-popup controls-FilterViewPanel__additional-editors-in-column"
        tabindex="0"
      >
        <div
          class="controls-FilterViewPanel__additional_left-column controls-FilterViewPanel__additional_left-column-maxWidth"
          data-qa="controls-FilterViewPanel__additional_left-column"
        >
          <div
            class="controls-FilterViewPanel__additional-editor controls-FilterViewPanel__additional-editor_default controls-fontsize-m controls-text-filterExtendedItem controls-FilterViewPanel__additional-editor-popup_last controls-FilterViewPanel__additional-editor-popup-in-column"
            data-name="FilterViewPanel__additional-editor_withCaption"
            data-qa="FilterViewPanel__additional-editor"
            title="Метка для блока \\"Ещё можно отобрать\\""
          >
            Метка для блока "Ещё можно отобрать"
          </div>
        </div>
        <div
          class="controls-FilterViewPanel__additional_right-column"
          data-qa="controls-FilterViewPanel__additional_right-column"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Controls/filterPanelPopup:Sticky caption tests Тест меток редакторов Разный текст метки при переходе из блока "Еще можно отобрать" в блок "Отбираются" Редактор в блоке "Отбираются" 1`] = `
<div
  class="controls_filterPanel_theme-default  controls-FilterViewPanel-contrast   controls_filter_theme-default controls-FilterViewPanel controls-Scroll-containerBase_userContent"
  data-qa="controls-FilterPanelPopup__sticky_filterPanel"
  tabindex="0"
>
  <div
    class="controls-FilterViewPanel_viewMode-popup-vertical  controls-FilterViewPanel-contrast"
    data-qa="controls-FilterViewPanel__basic-editors"
    tabindex="0"
  >
    <div
      class="controls-PropertyGrid controls_propertyGrid_theme-default controls_list_theme-default controls_toggle_theme-default controls-PropertyGrid__itemsContainer_paddingTop-null controls-PropertyGrid__itemsContainer_paddingBottom-null controls-PropertyGrid__itemsContainer_paddingLeft-filterPanelPopup controls-PropertyGrid__itemsContainer_paddingRight-l controls-FilterViewPanel__propertyGrid"
      data-qa="FilterViewPanel__propertyGrid"
      tabindex="0"
    >
      <div
        class="controls-FilterViewPanel__group-wrapper controls-FilterViewPanel__group-wrapper_style-master"
        data-qa="FilterViewPanel__group-wrapper"
        item-key="group-withCaption"
        tabindex="0"
      >
        <div
          class="controls-FilterViewPanel__separator-content_container controls-FilterViewPanel__separator-content_container_popup ws-flexbox controls-FilterViewPanel__group controls-FilterViewPanel__group_height_viewMode-popup"
          data-qa="FilterViewPanel__group"
          tabindex="0"
        >
          <div
            class="controls-FilterViewPanel__group-content controls-FilterViewPanel__group-content_popup  controls-FilterViewPanel__group-content-align-left"
            tabindex="0"
            title="Метка для блока \\"Отбираются\\""
          >
            Метка для блока "Отбираются"
          </div>
        </div>
      </div>
      <div
        attr-data-qa="controls-FilterViewPanel__item_withCaption"
        class="controls-PropertyGrid__editor    controls-PropertyGrid__editor_spacingTop_default
                       controls-PropertyGrid__editor_spacingBottom_null controls-PropertyGrid__editor_spacingRight_default controls-PropertyGrid__editor_spacingLeft_default controls-FilterViewPanel__propertyGrid_itemTemplate
                controls-FilterViewPanel__propertyGrid_itemTemplate_viewMode-popup-vertical_last-without-extended-items"
        data-qa="controls-PropertyGrid__editor"
        item-key="property-grid-item-withCaption"
        tabindex="1"
      >
        <div
          class="controls-PropertyGrid__editor-withoutCaption controls-FilterViewPanel__baseEditor-container controls-FilterViewPanel__baseEditor-container_default"
          data-qa="controls-PropertyGrid__editor_withCaption"
        >
          <span />
          <div
            class="controls-FilterViewPanel__baseEditor-cross_container"
          >
            <span
              class="controls-icon_size-FilterGroupIcon  controls-FilterViewPanel__groupReset-icon icon-CloseNew controls-icon"
              data-qa="FilterViewPanel__baseEditor-cross"
              title="Сбросить"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
